test/cppsuite/tests/cursor_bound_01.cpp

```diff
@@ -722,23 +710,27 @@ class cursor_bound_01 : public test {
                 if (ret == 0) {
                     const char *range_srch_key;
                     testutil_check(range_cursor->get_key(range_cursor.get(), &range_srch_key));
-
-                    range_cursor->set_key(range_cursor.get(), range_srch_key);
+                    std::string range_key_copy(range_srch_key);
+                    range_cursor->set_key(range_cursor.get(), range_key_copy.c_str());
                     ret = range_cursor->search(range_cursor.get());
 
-                    testutil_assert(ret == 0 || ret == WT_NOTFOUND);
+                    testutil_assert(ret == 0 || ret == WT_NOTFOUND || ret == WT_ROLLBACK);
+                    if (ret == WT_ROLLBACK)
+                        continue;
                     validate_bound_search(
-                      ret, range_cursor, srch_key, bound_pair.first, bound_pair.second);
+                      ret, range_cursor, range_key_copy, bound_pair.first, bound_pair.second);
                 }
 
                 tc->txn.add_op();
-                tc->txn.try_rollback();
+                tc->txn.commit();
```

Comment:
Do we need to call `add_op` if we call `commit`? Doing this seems like the number of operations does not matter.

Reply:
[e46764d](https://github.com/wiredtiger/wiredtiger/pull/8108/commits/e46764d67a603ad254d4727e0d4e0c9088ab4635)